src/bentoml/_internal/frameworks/diffusers.py

```diff
@@ -609,25 +789,93 @@ def _replace_scheduler(self, scheduler_txt: str):
                     error_message="cannot import scheduler class",
                 )
 
+    if support_lora:
+
+        def _load_lora_weights(
+            self: DiffusersRunnable,
+            lora_weights: LoraOptionType | list[LoraOptionType],
+        ):
+            _load_lora_weights_to_pipeline(self.pipeline, lora_weights, self.lora_dir)
+
+        def _unload_lora_weights(
+            self: DiffusersRunnable,
+        ):
+            self.pipeline.unload_lora_weights()
+
+            # clear cached lora weights from GPU memory
+            torch.cuda.empty_cache()
+
+    else:
+
+        def _load_lora_weights(
+            self: DiffusersRunnable,
+            lora_args: LoraOptionType | list[LoraOptionType],
+        ):
+            raise NotImplementedError(
+                f"Class {pipeline_class} is not a subclass of LoraLoaderMixin, cannot load lora weights"
+            )
+
+        def _unload_lora_weights(
+            self: DiffusersRunnable,
+        ):
+            raise NotImplementedError(
+                f"Class {pipeline_class} is not a subclass of LoraLoaderMixin, cannot unload lora weights"
+            )
```

Comment:
```suggestion
    def _load_lora_weights(
        self: DiffusersRunnable,
        lora_weights: LoraOptionType | list[LoraOptionType],
    ):
    	if not support_lora:
    		raise NotImplementedError(
                f"Class {pipeline_class} is not a subclass of LoraLoaderMixin, cannot load lora weights")
        _load_lora_weights_to_pipeline(self.pipeline, lora_weights, self.lora_dir)

	def _unload_lora_weights(self: DiffusersRunnable):
    	if not support_lora:
    		raise NotImplementedError(
                f"Class {pipeline_class} is not a subclass of LoraLoaderMixin, cannot unload lora weights")
        self.pipeline.unload_lora_weights()

        # clear cached lora weights from GPU memory
        torch.cuda.empty_cache()
```

Reply:
my way will avoid test the switch everytime in runtime (the method is defined separately at creation/compile time). But maybe this won't save too much cpu cycles. Let me think about it.

Reply:
I think this is purely bytecode optimization